tests/framework/artifacts.py

```diff
@@ -190,3 +234,50 @@ def artifacts(self):
     def __len__(self):
         """Return the artifacts array len."""
         return len(self._artifacts)
+
+
+class SnapshotType(Enum):
+    """Supported snapshot types."""
+
+    FULL = 0
+    DIFF = 1
+
+
+class Snapshot:
+    """Manages Firecracker snapshots."""
+
+    def __init__(self, mem, vmstate, disks, ssh_key):
+        """Initialize mem, vmstate, disks, key."""
+        assert mem is not None
+        assert vmstate is not None
+        assert disks is not None
+        assert ssh_key is not None
+        self._mem = mem
+        self._vmstate = vmstate
+        self._disks = disks
+        self._ssh_key = ssh_key
+
+    def rebase_snapshot(self, base):
```

Comment:
The assumption here is that only artifacts fetched from S3 are read-only, and when some of them need writing you just create a copy. Otherwise files created during tests are also writeable.

Reply:
Also we might revisit and might change this as we add more types of tests.